Tests/ApolloPaginationTests/GraphQLQueryPagerTests.swift

```diff
@@ -0,0 +1,301 @@
+
+    await pager.fetch()
+    await fulfillment(of: [serverExpectation])
+
+    var canLoadMore = await pager.canLoadPrevious
+    XCTAssertTrue(canLoadMore)
+
+    let secondPageExpectation = Mocks.Hero.ReverseFriendsQuery.expectationForPreviousItem(server: server)
+    let secondPageFetch = expectation(description: "Second Page")
+    secondPageFetch.expectedFulfillmentCount = 2
+    let subscription = await pager.subscribe(onUpdate: { _ in
+      secondPageFetch.fulfill()
+    })
+    try await pager.loadPrevious()
+    await fulfillment(of: [secondPageExpectation, secondPageFetch])
+    subscription.cancel()
+    canLoadMore = await pager.canLoadPrevious
+    XCTAssertFalse(canLoadMore)
+  }
+
+  @available(iOS 16.0, macOS 13.0, *)
+  func test_pager_cancellation_calls_callback() async throws {
+    server.customDelay = .milliseconds(1)
+    let pager = GraphQLQueryPager(pager: createForwardPager())
+    let serverExpectation = Mocks.Hero.FriendsQuery.expectationForFirstPage(server: server)
+
+    pager.fetch()
+    await fulfillment(of: [serverExpectation], timeout: 1)
+    server.customDelay = .milliseconds(200)
+    let secondPageExpectation = Mocks.Hero.FriendsQuery.expectationForSecondPage(server: server)
+    let callbackExpectation = expectation(description: "Callback")
+    pager.loadNext(completion: { _ in
+      callbackExpectation.fulfill()
+    })
+    try await Task.sleep(for: .milliseconds(50))
+    pager.cancel()
+    await fulfillment(of: [callbackExpectation, secondPageExpectation], timeout: 1)
+  }
+
+  @available(iOS 16.0, macOS 13.0, *)
+  func test_pager_cancellation_calls_callback_manyQueuedRequests() throws {
+    server.customDelay = .milliseconds(1)
+    let pager = GraphQLQueryPager(pager: createForwardPager())
+    let serverExpectation = Mocks.Hero.FriendsQuery.expectationForFirstPage(server: server)
+    var results: [Result<GraphQLQueryPager<ForwardQuery, ForwardQuery>.Output, Error>] = []
+    var errors: [PaginationError?] = []
+
+    pager.fetch()
+    wait(for: [serverExpectation], timeout: 1)
+    server.customDelay = .milliseconds(150)
+    pager.subscribe { result in
+      results.append(result)
+    }
+    let secondPageExpectation = Mocks.Hero.FriendsQuery.expectationForSecondPage(server: server)
+    pager.loadNext(completion: { error in
+      let error = try? XCTUnwrap(error as? PaginationError)
+      errors.append(error)
+    })
+    pager.loadNext(completion: { error in
+      let error = try? XCTUnwrap(error as? PaginationError)
+      errors.append(error)
+    })
+    DispatchQueue.global().asyncAfter(deadline: .now() + .milliseconds(10)) {
+        pager.cancel()
+    }
+
+    wait(for: [secondPageExpectation], timeout: 2)
+    XCTAssertEqual(results.count, 1) // once for original fetch
+    XCTAssertEqual(errors.count, 2)
+    XCTAssertTrue(errors.contains(where: { $0 == .cancellation }))
+  }
+
+  @available(iOS 16.0, macOS 13.0, *)
```

Comment:
@BobaFetters, the Tuist configuration work you're doing now should make it possible for us to remove these, right? We should be able to set it up so that the pagination test target has a higher deployment target configured, since the pagination SPM package has a higher minimum platform version anyways.